central/complianceoperator/v2/checkresults/datastore/datastore_impl.go

```diff
@@ -199,3 +189,19 @@ func (d *datastoreImpl) withCountByResultSelectQuery(q *v1.Query, countOn search
 	)
 	return cloned
 }
+
+func withSACFilter(ctx context.Context, targetResource permissions.ResourceMetadata, query *v1.Query) (*v1.Query, error) {
```

Comment:
Should put this function pkg so that others can use it if they need to do similar things?

Reply:
Also, worthwhile to have a sac expert check this

Reply:
The schema does have the correct `Compliance` resource.  But the best I could tell, since I'm building an object of `ResourceCountByResultByCluster` and not the storage object itself, it was missing that information.  So when I initially tested it without this filter, I was getting results I should not have.